jetty-http/src/test/java/org/eclipse/jetty/http/HttpURITest.java

```diff
@@ -361,6 +361,13 @@ public void testCanonicalDecoded()
                 {".%2e", "..", EnumSet.of(Ambiguous.SEGMENT)},
                 {"%2e%2e", "..", EnumSet.of(Ambiguous.SEGMENT)},
 
+                // empty segment treated as ambiguous
+                {"/foo//bar", "/foo//bar", EnumSet.of(Ambiguous.SEGMENT)},
+                {"/foo//../bar", "/foo/bar", EnumSet.of(Ambiguous.SEGMENT)},
+                {"/foo///../../../bar", "/bar", EnumSet.of(Ambiguous.SEGMENT)},
+                {"/foo/./../bar", "/bar", EnumSet.noneOf(Ambiguous.class)},
+                {"/foo//./bar", "/foo//bar", EnumSet.of(Ambiguous.SEGMENT)},
+
```

Comment:
Perhaps also test URIs like: `foo/bar`, `foo;/bar`, `;/bar`, `;?n=v`, `?n=v`
The last three of those have empty segments